### ISSUE TITLE
[V2] Click to filter in the top bar

### ISSUE THREAD

**Issue by krishnaglick** [airbytehq/airbyte#23635]
Figma's linked in [the epic](https://github.com/airbytehq/airbyte/issues/23345)

Blocked by https://github.com/airbytehq/airbyte/issues/23633

Based on the work in [Basic Top Bar](https://github.com/airbytehq/airbyte/issues/23633), clicking a stream card should filter the streams list to the streams in that state.

**Comment by krishnaglick** [comment]
Since streams are only ever in one state I'm deferring this work to v2, when we'll have streams in multiple states.

**Comment by chandlerprall** [comment]
superseded by updated designs & what we want to communicate to users in that space
